core/src/main/java/io/dekorate/utils/Metadata.java

```diff
@@ -54,6 +55,40 @@
     return Optional.empty();
   }
 
+  public static boolean addToLabels(Builder builder, String key, String value) {
+    try {
+      Method editMethod = builder.getClass().getMethod("editOrNewMetadata");
```

Comment:
Do we really need reflection here?

Reply:
Unfortunately, I have no good way of visiting a top-level resource builder and being able to access the metadata. This is possibly a limitation of the model (unless I miss something myself). But for now, we are stuck with reflection.